readthedocs/core/views/__init__.py

```diff
@@ -176,7 +176,9 @@ def resolve_404_path(project, version_slug=None, language=None):
 
         version = None
         if version_slug:
-            version = project.versions.get(slug=version_slug)
+            version_qs = project.versions.filter(slug=version_slug)
+            if version_qs.exists():
+                version = version_qs.first()
```

Comment:
This block could just be:

    version = project.versions.filter(slug=version_slug).first()

Reply:
Returns the first object matched by the queryset, or None if there is no matching object. 

https://docs.djangoproject.com/en/1.11/ref/models/querysets/#first